wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Legalhold.swift

```diff
@@ -76,26 +81,31 @@ class ZMConversationTests_Legalhold: ZMConversationTestsBase {
         }
     }
 
-    func testThatLegalholdIsDeactivatedInConversation_OnlyWhenTheLastLegalholdClientIsDeleted() {
-        syncMOC.performGroupedBlock {
-            // GIVEN
+    func testThatLegalholdIsDeactivatedInConversation_OnlyWhenTheLastLegalholdClientIsDeleted() async {
+        // GIVEN
+        var legalHoldClient: UserClient!
```

Comment:
Not sure if it makes any difference between using `syncMOC.performAndWait` and `await syncMOC.perform`.
But if you want to change it, here is a suggestion:

```
    func testThatLegalholdIsDeactivatedInConversation_OnlyWhenTheLastLegalholdClientIsDeleted() async {
        // GIVEN
        let (legalHoldClient, conversation) = await syncMOC.perform {
            let selfUser = ZMUser.selfUser(in: self.syncMOC)
            let otherUser = ZMUser.insertNewObject(in: self.syncMOC)

            self.createSelfClient(onMOC: self.syncMOC)
            self.createClient(ofType: .permanent, class: .phone, for: otherUser)
            let legalHoldClient = self.createClient(ofType: .legalHold, class: .legalHold, for: otherUser)

            let conversation = self.createConversation(in: self.syncMOC)
            conversation.conversationType = .group
            conversation.addParticipantsAndUpdateConversationState(users: Set([selfUser, otherUser]), role: nil)

            XCTAssertEqual(conversation.legalHoldStatus, .pendingApproval)
            return (legalHoldClient, conversation)
        }

        // WHEN
        await legalHoldClient.deleteClientAndEndSession()

            // THEN
        await syncMOC.perform {
            XCTAssertEqual(conversation.legalHoldStatus, .disabled)
        }
    }
```